### PR TITLE
Adding Github Actions

### PR DESCRIPTION
**Pull request by lordkyzr**
This PR adds support for Github actions for MacOS, Ubuntu, and Windows and various versions of dotnet core as well as .NET Framework

### CODE REVIEW

**Comment by lordkyzr** [comment]
When this is merged its extremely important its rebased 😆

File: .github/workflows/test_dotnet_core_versions.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Test Dotnet SDK Versions
+
+on: [pull_request, push, workflow_dispatch]
```
Comment (CMoncur) [inline .github/workflows/test_dotnet_core_versions.yml:3]: We can be more specific at this point about when we want the actions to run. The method I used to accomplish this was:

```yaml
on:
    push:
        branches: 
            - 'master'
            - '*-DEV'
    pull_request:
        branches:
            - 'master'
            - '*-DEV'
```

**Review by CMoncur — COMMENTED** [review]
It's probably a good idea to remove the Travis CI files at this point too 👍

File: .github/workflows/test_dotnet_core_versions.yml (hunk 1 above)
Comment (lordkyzr) [reply .github/workflows/test_dotnet_core_versions.yml]: Alrighty this was changed

**Review by bporter125 — APPROVED** [review]

**Review by jnpblrdrgz — APPROVED** [review]

**Review by CMoncur — APPROVED** [review]
:shipit:

**Pull request merged**